File: src/model_v4/entities/SBS_KONFIG.ts
import { Column, Entity, Index, JoinColumn, ManyToOne, OneToMany, PrimaryColumn } from "typeorm";
import { SBS_HW } from "./SBS_HW";
import { SBS_HWTYP } from "./SBS_HWTYP";


@Entity("SBS_KONFIG",{schema:"sbsdb"})
@Index("sbskonfig_index2",["BEZEICHNUNG",])
@Index("sbskonfig_index1",["HERSTELLER",])
@Index("FK415F10F55EEAD941",["hWTYP_INDEX",])
export class SBS_KONFIG {

    @PrimaryColumn("bigint",{ 
        nullable:false,
        primary:true,
        name:"KONFIG_INDEX"
        })
    KONFIG_INDEX:string;
        

    @Column("varchar",{ 
        nullable:false,
        length:50,
        name:"BEZEICHNUNG"
        })
    BEZEICHNUNG:string;
        

    @Column("varchar",{ 
        nullable:true,
        length:50,
        default: () => "'NULL'",
        name:"HD"
        })
    HD:string | null;
        

    @Column("varchar",{ 
        nullable:false,
        length:50,
        name:"HERSTELLER"
        })
    HERSTELLER:string;
        

    @Column("varchar",{ 
        nullable:true,
        length:50,
        default: () => "'NULL'",
        name:"PROZESSOR"
        })
    PROZESSOR:string | null;
        

    @Column("varchar",{ 
        nullable:true,
        length:50,
        default: () => "'NULL'",
        name:"RAM"
        })
    RAM:string | null;
        

    @Column("longtext",{ 
        nullable:true,
        default: () => "'NULL'",
        name:"SONST"
        })
    SONST:string | null;
        

    @Column("varchar",{ 
        nullable:true,
        length:50,
        default: () => "'NULL'",
        name:"VIDEO"
        })
    VIDEO:string | null;
        

   
    @ManyToOne(type=>SBS_HWTYP, SBS_HWTYP=>SBS_HWTYP.sBS_KONFIGs,{ onDelete: 'RESTRICT',onUpdate: 'RESTRICT' })
    @JoinColumn({ name:'HWTYP_INDEX'})
    hWTYP_INDEX:SBS_HWTYP | null;


   
    @OneToMany(type=>SBS_HW, SBS_HW=>SBS_HW.kONFIG_INDEX,{ onDelete: 'RESTRICT' ,onUpdate: 'RESTRICT' })
    sBS_HWs:SBS_HW[];
    
}

File: SbsdbServer/sbsdbv4/model/SbsKonfig.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace hb.SbsdbServer.sbsdbv4.model {
    [Table("SBS_KONFIG")]
    public class SbsKonfig {
        public SbsKonfig() {
            SbsHw = new HashSet<SbsHw>();
        }

        [Column("KONFIG_INDEX", TypeName = "bigint(20)")]
        public long KonfigIndex { get; set; }

        [Required]
        [Column("BEZEICHNUNG", TypeName = "varchar(50)")]
        public string Bezeichnung { get; set; }

        [Column("HD", TypeName = "varchar(50)")]
        public string Hd { get; set; }

        [Required]
        [Column("HERSTELLER", TypeName = "varchar(50)")]
        public string Hersteller { get; set; }

        [Column("PROZESSOR", TypeName = "varchar(50)")]
        public string Prozessor { get; set; }

        [Column("RAM", TypeName = "varchar(50)")]
        public string Ram { get; set; }

        [Column("SONST", TypeName = "longtext")]
        public string Sonst { get; set; }

        [Column("VIDEO", TypeName = "varchar(50)")]
        public string Video { get; set; }

        [Column("HWTYP_INDEX", TypeName = "bigint(20)")]
        public long? HwtypIndex { get; set; }

        [ForeignKey("HwtypIndex")]
        [InverseProperty("SbsKonfig")]
        public virtual SbsHwtyp HwtypIndexNavigation { get; set; }

        [InverseProperty("KonfigIndexNavigation")]
        public virtual ICollection<SbsHw> SbsHw { get; set; }
    }
}
